### PR TITLE
feat(js): inbox load css with the link element in header

### PR DESCRIPTION
**Pull request by LetItRock**
### What changed? Why was the change needed?
Load the Inbox default styles with the `link` element:
- during the local development we are using tsup with postcss to build the `index.css` file which is then served with `http-server`
- during the production build we are creating the jsdelivr link to cdn

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: packages/js/package.json [hunk 1]
```diff
@@ -31,13 +31,13 @@
   "private": true,
   "scripts": {
     "clean": "rimraf ./dist",
-    "start": "pnpm run build -- --watch --sourcemap",
-    "pre:build": "cp ./src/ui/index.css ./src/ui/index.directcss",
-    "build": "pnpm run clean && pnpm run pre:build && tsup && pnpm run post:build",
+    "start:server": "http-server ./dist -p 4010",
```
Comment (LetItRock) [inline packages/js/package.json:34]: http-server will serve the `index.css` file with a URL like `http://localhost:4010/index.css`

File: packages/js/package.json [hunk 2]
```diff
@@ -31,13 +31,13 @@
...
+    "build": "pnpm run clean && NODE_ENV=production tsup && pnpm run post:build",
```
Comment (LetItRock) [inline packages/js/package.json:35]: the build sets the NODE_ENV to `production` to create the jsdelivr url

File: packages/js/postcss.config.js [hunk 3]
```diff
@@ -2,5 +2,8 @@ module.exports = {
   plugins: {
     autoprefixer: {},
     tailwindcss: {},
+    cssnano: {
+      preset: 'default',
+    }
```
Comment (LetItRock) [inline packages/js/postcss.config.js:7]: minify css

File: packages/js/src/ui/novuUI.tsx [hunk 4]
```diff
@@ -6,10 +6,18 @@ import { Appearance } from './context';
...
-//@ts-expect-error inline import esbuild syntax
-import css from 'directcss:./index.directcss';
 import { InboxProps } from './components';
 
+// eslint-disable-next-line
+// @ts-ignore
+const isDev = __DEV__;
+// eslint-disable-next-line
+// @ts-ignore
+const version = PACKAGE_VERSION;
```
Comment (LetItRock) [inline packages/js/src/ui/novuUI.tsx:16]: these variables are set with TSUP define prop

File: packages/js/src/ui/novuUI.tsx [hunk 5]
```diff
@@ -6,10 +6,18 @@ import { Appearance } from './context';
...
+const cssHref = isDev
+  ? 'http://localhost:4010/index.css'
+  : `https://cdn.jsdelivr.net/npm/@novu/js@${version}/dist/index.css`;
```
Comment (LetItRock) [inline packages/js/src/ui/novuUI.tsx:19]: build the URL to the CSS file

File: packages/js/tsup.config.ts [hunk 6]
```diff
@@ -13,6 +14,14 @@ const processCSS = async (css: string, filePath: string) => {
   return result.css;
 };
 
+const buildCSS = async () => {
+  const cssFilePath = path.join(__dirname, './src/ui/index.css');
+  const destinationCssFilePath = path.join(__dirname, './dist/index.css');
+  const css = fs.readFileSync(cssFilePath, 'utf-8');
+  const processedCss = await processCSS(css, cssFilePath);
+  fs.writeFileSync(destinationCssFilePath, processedCss);
+};
```
Comment (LetItRock) [inline packages/js/tsup.config.ts:23]: compile css with postcss

**Review by SokratisVidros — APPROVED** [review]
Nice!!! How much do we save in bundle size by doing this? Is there an FOUC due to this? Should we add some [resource hints](https://web.dev/learn/performance/resource-hints) to speed up CSS loading?

File: packages/js/package.json [hunk 7]
```diff
@@ -61,10 +61,13 @@
     "bytes-iec": "^3.1.1",
     "chalk": "^5.3.0",
     "compression-webpack-plugin": "^10.0.0",
+    "concurrently": "^5.3.0",
+    "cssnano": "^7.0.4",
     "esbuild-plugin-compress": "^1.0.1",
     "esbuild-plugin-inline-import": "^1.0.4",
```
Comment (BiswaViraj) [inline packages/js/package.json:67]: ✏ **suggestion:** ‏We can remove `esbuild-plugin-inline-import`, as it is no longer used

**Review by BiswaViraj — APPROVED** [review]

**Comment by LetItRock** [comment]
> Nice!!! How much do we save in bundle size by doing this? Is there an FOUC due to this? Should we add some [resource hints](https://web.dev/learn/performance/resource-hints) to speed up CSS loading?

~23kb, yes I was thinking about that when went to sleep :D

File: packages/js/src/ui/components/Renderer.tsx [hunk 8]
```diff
@@ -54,10 +54,11 @@ export const Renderer = (props: RendererProps) => {
...
 
-    const styleEl = document.createElement('style');
-    styleEl.id = id;
-    document.head.insertBefore(styleEl, document.head.firstChild);
-    styleEl.innerHTML = props.defaultCss;
+    const link = document.createElement('link');
+    link.id = id;
+    link.rel = 'stylesheet';
+    link.href = props.cssHref;
+    document.head.insertBefore(link, document.head.firstChild);
```
Comment (desiprisg) [inline packages/js/src/ui/components/Renderer.tsx:61]: 💯

**Review by desiprisg — APPROVED** [review]

**Comment by desiprisg** [comment]
> ~23kb, yes I was thinking about that when went to sleep :D

Did it actually get that big? Didn't expect that it always seemed so small.

**Pull request merged**